client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -308,7 +308,8 @@ private boolean shouldAcquireSegment() throws ReaderNotInReaderGroupException {
             if (!state.isReaderOnline(readerId)) {
                 throw new ReaderNotInReaderGroupException(readerId);
             }
-            if (acquireTimer.hasRemaining()) {
+            // Acquire segments if no segments assigned to a reader and there are unassigned segments that can be acquired.
+            if ((!(state.getSegments(readerId).isEmpty() && state.getNumberOfUnassignedSegments() != 0)) && acquireTimer.hasRemaining()) {
```

Comment:
We already cover the case of unassigned segments == 0 below. This can be simplified to
`!state.getSegments(readerId).isEmpty() && acquireTimer.hasRemaining()`
But I'm not sure this is what we actually want to do. For example notice that acquireTimer gets updated with `UPDATE_WINDOW` to perform the update below. This could result in multiple threads from the same reader all trying to acquire segments at the same time. 

What is the problem that prompted this change?

Reply:
Re-verified the need for this change using an integration test, it is not required.
(The current `UPDATE_WINDOW` has a duration of 30 seconds)